core/src/autogluon/core/_setup_utils.py

```diff
@@ -17,9 +17,9 @@
 # Only put packages here that would otherwise appear multiple times across different module's setup.py files.
 DEPENDENT_PACKAGES = {
     "boto3": ">=1.10,<2",  # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
-    "numpy": ">=1.21,<1.27",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
+    "numpy": ">=1.21,<1.29",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
     "pandas": ">=2.0.0,<2.2.0",  # "<{N+1}" upper cap
-    "scikit-learn": ">=1.3.0,<1.5",  # "<{N+1}" upper cap
+    "scikit-learn": ">=1.3.0,<1.4",  # "<{N+1}" upper cap
```

Comment:
Keep `<1.5` and change comment to `"<{N+2}" upper cap`

scikit-learn follows deprecation strategy where it won't introduce breaking changes between minor versions without warnings, so N+2 is safe.

Reply:
Got it